itests/kit/node_unmanaged.go

```diff
@@ -24,3 +60,730 @@ type TestUnmanagedMiner struct {
+
+	minerAddrBytes := new(bytes.Buffer)
+	req.NoError(tm.ActorAddr.MarshalCBOR(minerAddrBytes))
+
+	head, err = tm.FullNode.ChainHead(ctx)
+	req.NoError(err)
+
+	tm.t.Logf("Miner %s: Fetching seed randomness for sector %d...", tm.ActorAddr, sectorNumber)
+	rand, err := tm.FullNode.StateGetRandomnessFromBeacon(ctx, crypto.DomainSeparationTag_InteractiveSealChallengeSeed, seedRandomnessHeight, minerAddrBytes.Bytes(), head.Key())
+	req.NoError(err)
+	seedRandomness := abi.InteractiveSealRandomness(rand)
+
+	tm.t.Logf("Miner %s: Obtained seed randomness for sector %d: %x", tm.ActorAddr, sectorNumber, seedRandomness)
+	return seedRandomness
+}
+
+func (tm *TestUnmanagedMiner) generateProveCommit(
+	ctx context.Context,
+	sectorNumber abi.SectorNumber,
+	proofType abi.RegisteredSealProof,
+	seedRandomness abi.InteractiveSealRandomness,
+	pieces []abi.PieceInfo,
+) []byte {
+	tm.t.Logf("Miner %s: Generating proof type %d Sector Proof for sector %d...", tm.ActorAddr, proofType, sectorNumber)
+	req := require.New(tm.t)
+
+	actorIdNum, err := address.IDFromAddress(tm.ActorAddr)
+	req.NoError(err)
+	actorId := abi.ActorID(actorIdNum)
+
+	tm.t.Logf("Miner %s: Running proof type %d SealCommitPhase1 for sector %d...", tm.ActorAddr, proofType, sectorNumber)
+
+	scp1, err := ffi.SealCommitPhase1(
+		proofType,
+		tm.sealedCids[sectorNumber],
+		tm.unsealedCids[sectorNumber],
+		tm.cacheDirPaths[sectorNumber],
+		tm.sealedSectorPaths[sectorNumber],
+		sectorNumber,
+		actorId,
+		tm.sealTickets[sectorNumber],
+		seedRandomness,
+		pieces,
+	)
+	req.NoError(err)
+
+	tm.t.Logf("Miner %s: Running proof type %d SealCommitPhase2 for sector %d...", tm.ActorAddr, proofType, sectorNumber)
+
+	sectorProof, err := ffi.SealCommitPhase2(scp1, sectorNumber, actorId)
+	req.NoError(err)
+
+	tm.t.Logf("Miner %s: Got proof type %d sector proof of length %d for sector %d", tm.ActorAddr, proofType, len(sectorProof), sectorNumber)
+
+	return sectorProof
+}
+
+func (tm *TestUnmanagedMiner) submitMessage(
+	ctx context.Context,
+	params cbg.CBORMarshaler,
+	value uint64,
+	method abi.MethodNum,
+) *api.MsgLookup {
+	enc, aerr := actors.SerializeParams(params)
+	require.NoError(tm.t, aerr)
+
+	tm.t.Logf("Submitting message for miner %s with method number %d", tm.ActorAddr, method)
+
+	m, err := tm.FullNode.MpoolPushMessage(ctx, &types.Message{
+		To:     tm.ActorAddr,
+		From:   tm.OwnerKey.Address,
+		Value:  types.FromFil(value),
+		Method: method,
+		Params: enc,
+	}, nil)
+	require.NoError(tm.t, err)
+
+	tm.t.Logf("Pushed message with CID: %s for miner %s", m.Cid(), tm.ActorAddr)
+
+	msg, err := tm.FullNode.StateWaitMsg(ctx, m.Cid(), 2, api.LookbackNoLimit, true)
+	require.NoError(tm.t, err)
+	tm.t.Logf("Message with CID: %s has been confirmed on-chain for miner %s", m.Cid(), tm.ActorAddr)
+
+	return msg
+}
+
+func requireTempFile(t *testing.T, fileContentsReader io.Reader, size uint64) *os.File {
+	// Create a temporary file
+	tempFile, err := os.CreateTemp("", "")
```

Comment:
these don't get cleaned up by the testing infra because they're not done inside a temp dir made with `t.TempDir`; I think we either need to pass in a temp dir, or make a new one here with `t.TempDir` and put it inside of that so we get proper cleanup